xdsl/dialects/fsm.py

```diff
@@ -200,11 +209,11 @@ def verify_(self):
             if isinstance(parent, Machine):
                 if not (
                     [operand.type for operand in self.operands]
-                    == [type(result) for result in parent.function_type.outputs]
+                    == [result for result in parent.function_type.outputs]
                     and len(self.operands) == len(parent.function_type.outputs)
                 ):
                     raise VerifyException(
-                        "Output types must be consistent with the machine's"
+                        "Output type must be consistent with the machine's"
```

Comment:
```suggestion
                        "Output type must match with the containing state machine's output types"
```

I know that the doc says "consistent", but that means to be the same/matching, right?
If so, reword messages below too?

Reply:
correct, it basically needs to be the same type as the output type of the function defined in the MachineOp